### PR TITLE
Export api through client

### PR DESCRIPTION
**Pull request by ssanjay1**
Exporting all @osdk/api entities through @osdk/client

### CODE REVIEW

File: packages/foundry-sdk-generator/src/generate/betaClient/generatePackage.ts [hunk 1]
```diff
@@ -124,7 +118,7 @@ export async function generatePackage(
       packageName: options.packageName,
       packagePath,
       packageVersion: options.packageVersion,
-      dependencies: resolvedDependencies,
+      dependencies: [],
```
Comment (ssanjay1) [inline packages/foundry-sdk-generator/src/generate/betaClient/generatePackage.ts:121]: Removed api as a direct dep

File: packages/generator/src/v2.0/generatePerQueryDataFiles.ts [hunk 2]
```diff
@@ -116,8 +120,9 @@ async function generateV2QueryFile(
   await fs.writeFile(
     path.join(outDir, `${query.shortApiName}.ts`),
     await formatTs(`
-        import type { QueryDefinition , VersionBound} from "@osdk/api";
-        import type { QueryParam, QueryResult } from "@osdk/api";
+        import type { QueryDefinition , QueryParam, QueryResult, VersionBound} from "${
```
Comment (ssanjay1) [inline packages/generator/src/v2.0/generatePerQueryDataFiles.ts:123]: I'm not sure why these were split up before, just combined. If this was done for a reason lmk and I'll revert

File: packages/client.test.ontology/generateMockOntology.js [hunk 3]
```diff
@@ -57,4 +57,8 @@ await generateClientSdkVersionTwoPointZero(
   },
   outDir,
   "module",
+  undefined,
+  undefined,
+  undefined,
+  true,
```
Comment (ssanjay1) [inline packages/client.test.ontology/generateMockOntology.js:63]: Use internal flag so we avoid circular deps on this one

File: etc/client.report.api.md [hunk 4]
```diff
@@ -4,10 +4,10 @@
 
 ```ts
 
-import type { ActionDefinition } from '@osdk/api';
+import { ActionDefinition } from '@osdk/api';
 import { ActionEditResponse } from '@osdk/api';
-import type { ActionMetadata } from '@osdk/api';
-import type { ActionParam } from '@osdk/api';
+import { ActionMetadata } from '@osdk/api';
+import { ActionParam } from '@osdk/api';
```
Comment (ericanderson) [inline etc/client.report.api.md:10]: Why don't these say `type` any more?

File: packages/generator/src/v2.0/generatePerQueryDataFiles.ts (hunk 2 above)
Comment (ericanderson) [reply packages/generator/src/v2.0/generatePerQueryDataFiles.ts:123]: Oversight probably

**Review by ericanderson — APPROVED** [review]
Awesome thank you

File: etc/client.report.api.md (hunk 4 above)
Comment (ssanjay1) [reply etc/client.report.api.md:10]: Mmm, not sure, I'll check

**Pull request merged**